test/test_function_cache_decorator.py

```diff
@@ -0,0 +1,207 @@
+from datetime import timedelta
+import time
+from dcicutils.function_cache_decorator import function_cache
+
+
+def test_function_cache_decorator():
+
+    called = 0
+
+    @function_cache
+    def f(n):
+        nonlocal called
+        called += 1
+        if n < 0:
+            return None
+        return n * n
+
+    assert f(3) == 9   # miss
+    assert f.cache_info()["size"] == 1
+    assert f(4) == 16  # miss
+    assert f.cache_info()["size"] == 2
+    assert f(5) == 25  # miss
+    assert f.cache_info()["size"] == 3
+    assert f(3) == 9   # hit
+    assert f.cache_info()["size"] == 3
+    assert f(4) == 16  # hit
+    assert f.cache_info()["size"] == 3
+    assert f(5) == 25  # hit
+    assert f.cache_info()["size"] == 3
+    assert f(3) == 9   # hit
+    assert f.cache_info()["size"] == 3
+    assert f(4) == 16  # hit
+    assert f.cache_info()["size"] == 3
+    assert f(5) == 25  # hit
+    assert f.cache_info()["size"] == 3
+    assert called == 3
+    assert f.cache_info()["hits"] == 6
+    assert f.cache_info()["misses"] == 3
+    assert f.cache_info()["size"] == 3
```

Comment:
You might consider wrapping this in an object and making these properties so you don't need to index them like this

Reply:
Yes, I like that - namedtuple - thanks.